### ISSUE TITLE
DataTable: ALT+W Keyboard Shortcut to AUTO adjust the width of resizeable columns

### ISSUE THREAD

**Issue by Rene-Stoeckel** [primefaces/primefaces#11918]
### Description

We use the primefaces data table to display data and we use resizable columns so that the column widths
can be adjusted with the mouse.
In our accessibility assessment we were requested to make the columns resizable via keyboard as well.
For example we should provide a keyboard shortcut similiar to excel so that a user can 
auto-adjust the width of the column so that the cells content is fully visible at least to the width of the screen.

As a precondition, the feature (keyboard navigation on data tables)
https://github.com/primefaces/primefaces/issues/11783 must probably be implemented first.

### Describe the solution you would like

Provide a keyboard shortcut similiar to excel so that a user can auto-adjust the width of the column so that the cells content is fully visible at least to the width of the screen.

### Additional context

Preferably it would be the keyboard combination <kbd>ALT+W</kbd> tro trigger an autosize

**Comment by melloware** [comment]
@Rene-Stoeckel just for reference what are the keyboard shortcuts in Excel to do this?

**Comment by Rene-Stoeckel** [comment]
> @Rene-Stoeckel just for reference what are the keyboard shortcuts in Excel to do this?
In Excel its just a navigation through the menus using ALT->H->O->I (in german ALT->R->FF->I).
https://www.launchexcel.com/column-width-autofit-keyboard
It should be something simpler, like ALT+W etc.

**Comment by melloware** [comment]
Oh you want it to Auto Adjust the width of the column.

**Comment by melloware** [comment]
This is not trivial, I think this would require:

1. For the current column that has focus, find the longest text in the column by walking all the cells.
2. For that text try and determine a "visible pixel size" based on the current font size etc.
3. Change the current column width to be that new calculated "visible pixel size".

**Comment by melloware** [comment]
@Rene-Stoeckel please have your team test out this reproducer:
[pf-11918.zip](https://github.com/primefaces/primefaces/files/15336327/pf-11918.zip)

Unzip and run `mvn clean jetty:run` and navigate to: http://localhost:8080/

Now when you are in a column and press `ALT+W` it resizes teh column to fit the largest element.

![datatable-resizer](https://github.com/primefaces/primefaces/assets/4399574/89e8de0f-c0d4-4c18-b182-d36b3b0d3d59)

**Comment by Rene-Stoeckel** [comment]
We just tested this reproducer with the screenreader and we can confirm that 
1) the content of the table is correctly presented with the screenreader, just as expected.
2) cursor navigation is working as expected,
2) the column size can be adjusted as expected with ALT-W.

But there are things we noticed that still must be fixed here:
1) Somehow, when text in the table is marked with the mouse then often the cursor navigation is broken. The fokus travels to a different cell than expected and some cells can no longer be reached.
2) The auto resize of a column should also consider the length of the text its colmn header including icon so that the header would be fully visible after resize as well.

**Comment by melloware** [comment]
@Rene-Stoeckel for issue 1 can you reproduce it exactly?  or give me instructions on how to reproduce the issue?

as for Issue 2 let me see what I can do.

**Comment by Rene-Stoeckel** [comment]
I updated the description of finding 1) in the comment above to be more specific.

**Comment by melloware** [comment]
OK I think I fixed both issues if you want to try this one.  Don't forget to CTRL+F5 to reload JS.
[pf-dt-accessibility.zip](https://github.com/primefaces/primefaces/files/15349145/pf-dt-accessibility.zip)

**Comment by Rene-Stoeckel** [comment]
Thank you very much for the quick response. We tried this tester and the two issues are fixed. Screenreader, cursor navigation and column auto resize are as expeced.

**Comment by melloware** [comment]
OK i will put a PR together to get this incorporated into PF.  Thank you for the detailed testing!
